### DISCUSSION TITLE
Typer subcommand when called as function call from another subcommand doesn't take default typer option value, but it takes the typer object as parameter.

### DISCUSSION THREAD

**Discussion by pmaru10** [fastapi/typer#605] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer()


@app.command()
def cmd1(number: int = typer.Option(1, help="An integer for cmd1")):
    typer.echo(f"Running cmd1 with number: {number}")
    cmd2()

@app.command()
def cmd2(number: int = typer.Option(2, help="An integer for cmd2")):
    typer.echo(f"Running cmd2 with number: {number}")


if __name__ == "__main__":
    app()
~
```

### Description

Typer subcommand when called as a function call from another subcommand, the function doesn't take the default value as an argument but rather takes the typer object.

In the above sample snippet, i was expecting the output

```
$ python3 test.py cmd1
Running cmd1 with number: 1
Running cmd2 with number: 2
$ 
```

But, I am seeing the following output.

```
$ python3 test.py cmd1
Running cmd1 with number: 1
Running cmd2 with number: <typer.models.OptionInfo object at 0x7fa1a5c29a90>
$ 
```

Am I missing here something?

### Operating System

Linux

### Operating System Details

Ubuntu 20.04

### Typer Version

0.7.0

### Python Version

Python 3.8.10

**Comment by Jylpah** [comment]
Hi, 

I do not know how Typer works internally, but you called `cmd2()` so the default argument will be `typer.Option(2, help="An integer for cmd2")` as you have specified in the function definition. If you call `cmd2(2)` it should do what you expected. I _guess_ that Typer wrapper deal with the function parameter setting. So I do not see any bug here.

**Comment by YuriiMotov** [comment, marked as answer]
This is a disadvantage of old-style approach when you specify `` as default value.
To avoid this, use Annotated approach:

Instead of
```py
def cmd1(number: int = typer.Option(1, help="An integer for cmd1")):
```
use
```py
def cmd1(number: Annotated[int, typer.Option(help="An integer for cmd1")] = 1):
```

Full example:
```py
from typing import Annotated
import typer

app = typer.Typer()


@app.command()
def cmd1(number: Annotated[int, typer.Option(help="An integer for cmd1")] = 1):
    typer.echo(f"Running cmd1 with number: {number}")
    cmd2()

@app.command()
def cmd2(number: Annotated[int, typer.Option(help="An integer for cmd2")] = 2):
    typer.echo(f"Running cmd2 with number: {number}")


if __name__ == "__main__":
    app()
```
